### PR TITLE
Fix FLI/FLC decoder for files with a prefix chunk

### PR DESCRIPTION
**Pull request by twolife**
fix FLC decoder for files with prefix chunk

without the seek() the decoder continues to read bytes in the prefixchunk and have zero chance to find the 0xF1FA magic number ut needs just after

### CODE REVIEW

**Comment by hugovk** [comment]
Please could you add a test that fails without the fix, and passes with it?

**Comment by twolife** [comment]
This is reproducible with this sample : https://samples.ffmpeg.org/fli-flc/2422.FLC

```
from PIL import Image, ImageFile
ImageFile.LOAD_TRUNCATED_IMAGES = True
im = Image.open("2422.FLC")
im.seek(15)
im.save('frame15.png')
```
You get garbage without the patch and a clean image with the patch

**Comment by hugovk** [comment]
Can you write a unit test, with an image that can be contributed under Pillow's licence?

A unit test helps ensure this doesn't break in the future.

**Comment by twolife** [comment]
test added

**Review by radarhere — APPROVED** [review]

File: Tests/test_file_fli.py [hunk 1]
```diff
@@ -31,6 +34,22 @@ def test_sanity() -> None:
...
+            assert im.format == "FLI"
+            assert im.info["duration"] == 171
+            assert im.is_animated
+
+            palette = im.getpalette()
+            assert palette[3:6] == [255, 255, 255]
+            assert palette[381:384] == [204, 204, 12]
+            assert palette[765:] == [252, 0, 0]
+    finally:
+        ImageFile.LOAD_TRUNCATED_IMAGES = False
```
Comment (hugovk) [inline Tests/test_file_fli.py:51]: Please can you move this to its own test function?

File: Tests/test_file_fli.py [hunk 2]
```diff
@@ -32,6 +35,24 @@ def test_sanity() -> None:
         assert im.is_animated
 
 
+def test_flc_with_prefixchunk_sanity() -> None:
```
Comment (radarhere) [inline Tests/test_file_fli.py:38]:
```suggestion
def test_flc_with_prefix_chunk() -> None:
```
`test_sanity` is the name we use for a test method at the start that checks basic functionality. You don't need to use 'sanity' in any other method.

File: Tests/test_file_fli.py [hunk 3]
```diff
@@ -4,17 +4,20 @@
...
 # created as an export of a palette image from Gimp2.6
 # save as...-> hopper.fli, default options.
 static_test_file = "Tests/images/hopper.fli"
 
-# From https://samples.libav.org/fli-flc/
+# From https://samples.ffmpeg.org/fli-flc/
 animated_test_file = "Tests/images/a.fli"
 
+# From https://samples.ffmpeg.org/fli-flc/
+animated_flc_with_prefixchunk_test_file = "Tests/images/2422.flc"
```
Comment (radarhere) [inline Tests/test_file_fli.py:19]:
```suggestion
animated_flc_with_prefix_chunk_test_file = "Tests/images/2422.flc"
```

File: Tests/test_file_fli.py [hunk 4]
```diff
@@ -32,6 +35,24 @@ def test_sanity() -> None:
         assert im.is_animated
 
 
+def test_flc_with_prefixchunk_sanity() -> None:
+    ImageFile.LOAD_TRUNCATED_IMAGES = True
+    try:
+        with Image.open(animated_flc_with_prefixchunk_test_file) as im:
```
Comment (radarhere) [inline Tests/test_file_fli.py:41]:
```suggestion
        with Image.open(animated_flc_with_prefix_chunk_test_file) as im:
```

File: Tests/test_file_fli.py [hunk 5]
```diff
@@ -4,17 +4,20 @@
...
 # created as an export of a palette image from Gimp2.6
 # save as...-> hopper.fli, default options.
 static_test_file = "Tests/images/hopper.fli"
 
-# From https://samples.libav.org/fli-flc/
+# From https://samples.ffmpeg.org/fli-flc/
 animated_test_file = "Tests/images/a.fli"
 
+# From https://samples.ffmpeg.org/fli-flc/
+animated_flc_with_prefix_chunk_test_file = "Tests/images/2422.flc"
```
Comment (radarhere) [inline Tests/test_file_fli.py:19]:
```suggestion
animated_test_file_with_prefix_chunk = "Tests/images/2422.flc"
```

File: Tests/test_file_fli.py [hunk 6]
```diff
@@ -32,6 +35,24 @@ def test_sanity() -> None:
         assert im.is_animated
 
 
+def test_prefix_chunk() -> None:
+    ImageFile.LOAD_TRUNCATED_IMAGES = True
+    try:
+        with Image.open(animated_flc_with_prefix_chunk_test_file) as im:
```
Comment (radarhere) [inline Tests/test_file_fli.py:41]:
```suggestion
        with Image.open(animated_test_file_with_prefix_chunk) as im:
```

Comment (radarhere) [reply Tests/test_file_fli.py:41]: Yes, I know, you just changed this - as an explanation, I did edit my previous suggestion a second later, but I'm guessing you just read the email.

Also, you don't have to force push your changes each time. You can apply suggestions like this through the Github interface in the 'Files changed' tab if you want.

**Review by radarhere — APPROVED** [review]

**Comment by hugovk** [comment]
Thank you!

**Pull request merged**